source/slang/slang-check-conformance.cpp

```
@@ -66,35 +66,90 @@ SubtypeWitness* SemanticsVisitor::isSubtype(
-    if (auto fwdDiffFuncType = as<FwdDiffFuncType>(callableDeclRef.getDecl()->funcType.type))
+    auto rawDirectFuncType = callableDeclRef.getDecl()->funcType.type;
+    Type* substitutedDirectFuncType = rawDirectFuncType
+                                          ? dynamicCast<Type>(substituteType(
+                                                                  SubstitutionSet(callableDeclRef),
+                                                                  getCurrentASTBuilder(),
+                                                                  rawDirectFuncType)
+                                                                  ->resolve())
+                                          : nullptr;
+
+    if (auto rawFwdDiffFuncType = as<FwdDiffFuncType>(rawDirectFuncType))
     {
```

Cppcheck (2.20.0)

[information] Include file

(missingInclude)